include/crocoddyl/core/controls/poly-two-rk4.hpp

```diff
@@ -0,0 +1,68 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2021, LAAS-CNRS, University of Trento
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#ifndef CROCODDYL_CORE_CONTROLS_POLY_TWO_RK4_HPP_
+#define CROCODDYL_CORE_CONTROLS_POLY_TWO_RK4_HPP_
+
+#include "crocoddyl/core/fwd.hpp"
+#include "crocoddyl/core/utils/exception.hpp"
+#include "crocoddyl/core/control-base.hpp"
+
+namespace crocoddyl {
+
+/**
+ * @brief A polynomial function of time of degree two, that is a quadratic function
+ * 
+ * The size of the parameters p is 3 times the size of the control input u.
+ * The first third of p represents the value of u at time 0.
+ * The second third of p represents the value of u at time 0.5.
+ * The last third of p represents the value of u at time 1.
+ * This parametrization is suitable to be used with the RK-4 integration scheme, 
+ * because it requires the value of u exactly at 0, 0.5 and 1.
+ */
+template <typename _Scalar>
+class ControlPolyTwoRK4Tpl : public ControlAbstractTpl<_Scalar> {
```

Comment:
The reason why `RK4` is in the name is that, all the possible parametrization of a 2nd-order polynomial, this is the one that is best suited to be used with an RK4 integration scheme because the three parameters represent exactly the value of u at time 0, 0.5 and 1, which are the times used by RK4 (or better, I should say the RK4 that is implemented in Crocoddyl). In another branch I've implemented also RK3, and another 2nd-order polynomial parametrization that is more suited for RK3, therefore I think it's important that RK4 remains in the name of this class. 

Regarding the idea of using "quadratic" instead of "poly2", the same comment I've left above applies here.

Reply:
> The reason why `RK4` is in the name is that, all the possible parametrization of a 2nd-order polynomial, this is the one that is best suited to be used with an RK4 integration scheme because the three parameters represent exactly the value of u at time 0, 0.5 and 1, which are the times used by RK4 (or better, I should say the RK4 that is implemented in Crocoddyl). In another branch I've implemented also RK3, and another 2nd-order polynomial parametrization that is more suited for RK3, therefore I think it's important that RK4 remains in the name of this class.

Thanks for the explanation. I have a follow-up question here. Can we use `ControlPolyTwoRK4` in other integrated-action models? If so, what are the cons of such choice? Or better could you elaborate a bit more the statement *"best suited to be used with an RK4"*.

If this control can only be used with the RK4 integrated-action model. Then, we should include a rule to prevent users to do that. I could provide you a way of doing it (if this needed).



> Regarding the idea of using "quadratic" instead of "poly2", the same comment I've left above applies here.

Let's keep `Poly2` as this naming choice seems to be more conventional in this case.